nasa-gibs/worldview
# WV-2145: Mobile Animation Feature

## Description

This update allows for animation creation within mobile views. The mobile animation frame limit have been set to 50 frames to account for limited bandwidth on mobile devices. The animations in mobile views have limited features and do not include GIF creation. 

Added a mobile animation button which is limited to mobile views that brings up a new mobile animation widget. After selections on this widget are selected, it can be minimized and the animation can be played with the collapsed animation widget. 

Overhauled the conditional styling for the animation button, animation play button (collapsed animation widget) and date arrows so that they render in proper positions depending on breakpoints and subdaily mode. 

## How To Test

1. Launch WV
2. Open the DevTools
3. Select any mobile view
4. The new mobile animation button should appear which brings up the new mobile animation widget.
5. The mobile animation widget should work in any mobile view including smaller desktop views.
6. To test the animation frame limit, go to [localhost:3000/?v=-127.54084611130202,-31.196051270164425,-36.29880238885806,62.96630766900102&z=4&ics=true&ici=5&icd=10&as=2021-12-03-T16%3A00%3A00Z&ae=2021-12-03-T17%3A10%3A00Z&l=GOES-East_ABI_GeoColor,Coastlines_15m&lg=true&al=true&ab=on&t=2021-12-01-T20%3A10%3A00Z](url)
7. Open the animation widget and decrease the year by 1 to 2021.
8. Observe that the mobile animation play button to  play the animation does not appear when the frame limit is exceeded.
